File: Program/Tier1/Shared/DTOs/HostUpdateDTO.cs
namespace Shared.DTOs;

public class HostUpdateDTO
{
    public string Email { get; }
    public string? Password { get; }
    public char? Gender { get; }
    public string? Nationality { get; }
    public string? FirstName { get; }
    public string? MiddleName { get;  }
    public string? LastName { get; }

    public HostUpdateDTO(string email, string? password, char? gender, string? nationality, string? firstName, string? middleName, string? lastName)
    {
        Email = email;
        Password = password;
        Gender = gender;
        Nationality = nationality;
        FirstName = firstName;
        MiddleName = middleName;
        LastName = lastName;
    }
}

File: Program/Tier2/src/main/java/via/sep3/group11/tier2/shared/DTOs/HostUpdateDTO.java
package via.sep3.group11.tier2.shared.DTOs;

public class HostUpdateDTO {
    private String email;
    private String password;
    private char gender;
    private String nationality;
    private String firstName;
    private String middleName;
    private String lastName;

    public HostUpdateDTO(String email, String password, char gender, String nationality, String firstName, String middleName, String lastName) {
        this.email = email;
        this.password = password;
        this.gender = gender;
        this.nationality = nationality;
        this.firstName = firstName;
        this.middleName = middleName;
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public char getGender() {
        return gender;
    }

    public String getNationality() {
        return nationality;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getMiddleName() {
        return middleName;
    }

    public String getLastName() {
        return lastName;
    }
}
